homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,243 @@
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.template import Template
+
+from . import SimpliSafe, SimpliSafeEntity
+from .const import DOMAIN, LOGGER
+
+DEFAULT_IMAGE_WIDTH = 720
+
+# The Outdoor Camera has a custom service for saving the motion clip.
+ATTR_FILENAME = "filename"
+ATTR_WIDTH = "width"
+
+SERVICE_OC_IMAGE = "capture_motion_image"
+SERVICE_OC_CLIP = "capture_motion_clip"
+
+SERVICE_OC_IMAGE_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_FILENAME): cv.template,
+        vol.Optional(ATTR_WIDTH, default=720): vol.Coerce(int),
+    }
+)
+
+SERVICE_OC_CLIP_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_FILENAME): cv.template,
+    }
+)
+
+SERVICES = (
+    SERVICE_OC_IMAGE,
+    SERVICE_OC_CLIP,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SimpliSafe cameras based on a config entry."""
+    simplisafe = hass.data[DOMAIN][entry.entry_id]
+
+    cameras: list[SimplisafeOutdoorCamera] = []
+
+    for system in simplisafe.systems.values():
+        if system.version == 2:
+            LOGGER.info("Skipping camera setup for V2 system: %s", system.system_id)
+            continue
+
+        for camera in system.cameras.values():
+            if camera.camera_type == CameraTypes.OUTDOOR_CAMERA:
+                cameras.append(
+                    SimplisafeOutdoorCamera(
+                        hass,
+                        simplisafe,
+                        system,
+                        camera,
+                    )
+                )
+
+    async_add_entities(cameras)
+
+
+class SimplisafeMotionCamera(SimpliSafeEntity, Camera):
+    """A camera base class that supports motion capture media."""
+
+    _device: DeviceV3
+    _attr_image_last_updated: datetime | None = None
+    _attr_image_url: str | None = None
+    _attr_clip_url: str | None = None
+    _attr_hls_url: str | None = None
+    _attr_cached_image: bytes | None = None
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        simplisafe: SimpliSafe,
+        system: SystemV3,
+        camera: DeviceV3,
+    ) -> None:
+        """Initialize."""
+        SimpliSafeEntity.__init__(
+            self,
+            simplisafe,
+            system,
+            device=camera,
+            additional_websocket_events=[EVENT_CAMERA_MOTION_DETECTED],
+        )
+        Camera.__init__(self)
+
+        self._hass = hass
+        self._register_services()
+
+    @property
+    def name(self) -> str | None:
+        """Return a good name for this camera."""
+        return self._device.name
+
+    @callback
+    def async_unload(self) -> None:
```

Comment:
Where does this get called?

Reply:
Does HA call this at some point?  I believe I found this sort of thing in other components, so I adopted it here too.  If HA does not call it, then I'll remove it ... because I certainly don't call it.